src/main/java/com/google/cloud/genomics/dataflow/pipelines/LoadReadsToBigQuery.java

```diff
@@ -0,0 +1,377 @@
+                                 c.element().getNextMatePosition().getReverseStrand())
+               );
+      // Alignment
+      TableRow alignmentPosition =
+          new TableRow().set(REFERENCE_NAME,
+                             c.element().getAlignment().getPosition().getReferenceName())
+                        .set(POSITION, c.element().getAlignment().getPosition().getPosition())
+                        .set(REVERSE_STRAND,
+                             c.element().getAlignment().getPosition().getReverseStrand());
+      List<TableRow> alignmentCigars = new ArrayList<>();
+      for (CigarUnit cigar : c.element().getAlignment().getCigarList()) {
+        TableRow alignmentCigar =
+            new TableRow().set(OPERATION, cigar.getOperation().getNumber())
+                          .set(OPERATION_LENGTH, cigar.getOperationLength())
+                          .set(REFERENCE_SEQUENCE, cigar.getReferenceSequence());
+        alignmentCigars.add(alignmentCigar);
+      }
+      TableRow alignmentRecord = new TableRow().set(POSITION, alignmentPosition)
+          .set(MAPPING_QUALITY, c.element().getAlignment().getMappingQuality())
+          .set(CIGAR, alignmentCigars);
+      row.set(ALIGNMENT, alignmentRecord);
+
+      c.output(row);
+    }
+  }
+
+  public static void main(String[] args) throws GeneralSecurityException, IOException,
+         URISyntaxException {
+    // Register the options so that they show up via --help
+    PipelineOptionsFactory.register(Options.class);
+    pipelineOptions = PipelineOptionsFactory.fromArgs(args)
+        .withValidation().as(Options.class);
+    // Option validation is not yet automatic, we make an explicit call here.
+    Options.Methods.validateOptions(pipelineOptions);
+    // Needed for BigQuery.
+    pipelineOptions.setTempLocation(pipelineOptions.getTempPath());
+
+    auth = GenomicsOptions.Methods.getGenomicsAuth(pipelineOptions);
+    p = Pipeline.create(pipelineOptions);
+
+    // Ensure data is accessible
+    String BAMFilePath = pipelineOptions.getBAMFilePath();
+    if (!Strings.isNullOrEmpty(BAMFilePath)) {
+      if (GCSURLExists(BAMFilePath)) {
+        System.out.println(BAMFilePath + " is present, good.");
+      } else {
+        System.out.println("Error: " + BAMFilePath + " not found.");
+        return;
+      }
+      if (pipelineOptions.isShardBAMReading()) {
+        // The BAM code expects an index at BAMFilePath+".bai"
+        // and sharded reading will fail if the index isn't there.
+        String BAMIndexPath = BAMFilePath + ".bai";
+        if (GCSURLExists(BAMIndexPath)) {
+          System.out.println(BAMIndexPath + " is present, good.");
+        } else {
+          System.out.println("Error: " + BAMIndexPath + " not found.");
+          return;
+        }
+      }
+    }
+
+    PCollection<Read> reads = getReads();
+    // TODO(nmousavi): validate the given BigQuery table, and fail fast if it is not valid and empty.
```

Comment:
Better to leave TODOs in the form of issue references (so we can create an issue for this and include it here

TODO(#xx): Validate table existance.